src/gucs.rs

```diff
@@ -7,13 +7,31 @@ All rights reserved.
 Use of this source code is governed by the PostgreSQL license that can be found in the LICENSE.md file.
 */
 
+use once_cell::sync::Lazy;
 use pgx::guc::{GucContext, GucRegistry, GucSetting};
 use std::path::PathBuf;
 use std::str::FromStr;
 
 static PLRUST_WORK_DIR: GucSetting<Option<&'static str>> = GucSetting::new(None);
 static PLRUST_PG_CONFIG: GucSetting<Option<&'static str>> = GucSetting::new(None);
 static PLRUST_TRACING_LEVEL: GucSetting<Option<&'static str>> = GucSetting::new(None);
+pub(crate) static PLRUST_ALLOWED_DEPENDENCIES: GucSetting<Option<&'static str>> =
+    GucSetting::new(None);
+
+pub(crate) static PLRUST_ALLOWED_DEPENDENCIES_CONTENTS: Lazy<toml::value::Table> =
```

Comment:
I still didn't address the Lazy portion here with SIGHUP cause I wasn't sure if we wanted to handle the case where the file is the same, but the contents have changed or not?

Reply:
There are almost 100,000 crates on crates.io, so... I am not against reading it every time in principle, but I am not sure if it is actually a good idea to simply blindly open it each time.

Reply:
Actually I think the proper way to handle this would be, assuming it's possible, to open up the pgx guc api: https://github.com/tcdi/pgx/blob/master/pgx/src/guc.rs#L220

to expose GucStringCheckHook and other variants:
https://github.com/postgres/postgres/blob/f13b2088fa2d4455936e65459b77698a4452f932/src/include/utils/guc.h#L177-L181

And then we can define a function that when the hook is called, we validate the file exists, read it, and update our value. That would avoid us having to figure out when a SIGHUP is actually issued to re-read the file contents.

What do you think? As a (sad) workaround/TODO I could re-define it as PGC_POSTMASTER instead of PGC_SIGHUP since Eric's patch makes that possible now with it being loaded in shared_preload_libraries

Reply:
Yeah, I'm good with accepting a temporary fix for now and digging into doing it "properly" a bit later. Using the Guc hook sounds interesting/useful.

> Good point...I was thinking it'd be in the low 10s of crates that an admin would want.

Rust programmers tend to do an unusual amount of grepping and otherwise programmatically manipulating the entire source codebase of crates.io, so my inclination is to assume it's entirely plausible someone managing a PL/Rust instance will do crazy things like list every single crate minus ~100, instead.

Reply:
Also I should note that "do silly things" is precisely a case where I am not concerned about providing optimal performance, I just don't want to assume unwarrantedly that a number that could easily be between 0 and 100k will necessarily be only 1-50.

Reply:
Makes sense, thanks! I've updated it to PGC_Postmaster as a temporary workaround, we can come back to it when GucHooks are supported